### PR TITLE
Document Request save + create payment flow

### PR DESCRIPTION
**Pull request by lekshmimallika-aot**
*Issue #:* /bcgov/entity#

*Description of changes:*

Marking it as draft to complete the postman tests and unit tests of validator and post endpoint. Otherwise this is ready for review

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: search-api/src/search_api/services/payment.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+from requests import exceptions
+from flask import current_app
+from flask_jwt_oidc import JwtManager
+
+from search_api.exceptions import ApiConnectionException
+
+
+# Maps Document Type to Pay API Filing Type
+DOCUMENT_TYPE_TO_FILING_TYPE = {
+    'BUSINESS_SUMMARY_FILING_HISTORY': 'BSRCH'
```
Comment (kialj876) [inline search-api/src/search_api/services/payment.py:29]: Can you use the Document.DocumentType enum here or is there a reason not to?

File: search-api/src/search_api/resources/v1/businesses/document_request.py [hunk 2]
```diff
@@ -50,3 +54,33 @@ def get(business_identifier, request_id=None):
         return jsonify(documentAccessRequests=access_requests_list)
     except Exception as default_exception:  # noqa: B902
         return resource_utils.default_exception_response(default_exception)
+
+
+@bp.route('/<string:business_identifier>/documents/requests', methods=['POST', 'OPTIONS'])
```
Comment (kialj876) [inline search-api/src/search_api/resources/v1/businesses/document_request.py:59]: fyi I think you can simplify this to @bp.post(<route>) and then name the function below it something more meaningful if you want. Same with the gets above. Not actually sure if its better or not though haha - I did it for the search endpoints (I think its a new flask 2 thing?)

**Review by kialj876 — APPROVED** [review]
Hey Lekshmi, nice work! Couple questions, but lgtm

**Comment by lekshmimallika-aot** [comment]
Merging this for the queue publisher code to be added. Remaining tests will be in next PR

**Pull request merged**